.github/workflows/build.yaml

```diff
@@ -9,11 +9,39 @@ on:
       - master
 
 jobs:
-  hello:
-    name: Hello
+  static-analysis:
+    name: Static Analysis
 
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        php-version:
+          - 7.2
+
     steps:
-      - name: Echo a greeting
-        run: echo 'Hello, GitHub Actions!'
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          coverage: none
+          php-version: ${{ matrix.php-version }}
+
+      - name: Determine composer cache directory
+        id: determine-composer-cache-directory
+        run: echo "::set-output name=directory::$(composer config cache-dir)"
+
+      - name: Cache dependencies installed with composer
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.determine-composer-cache-directory.outputs.directory }}
+          key: php-${{ matrix.php-version }}-composer-${{ hashFiles('composer.json') }}
```

Comment:
Notice how we use the `${{ matrix.php-version }}` expression which resolves to the corresponding value in the build matrix. Here it will resolve to `7.2`.

Also, we use the `hashfiles()` function to calculate a hash from `composer.json` as part of the cache key.

💁‍♂️ For reference, see:

- https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix
- https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#hashfiles